core/modules/comment/comment.theme.inc

```diff
@@ -106,9 +106,16 @@ function template_preprocess_comment(&$variables) {
   $uri['options'] += array('attributes' => array('class' => array('permalink'), 'rel' => 'bookmark'));
 
   $variables['title'] = l($comment->subject, $uri['path'], $uri['options']);
+  // In case title is hidden, we can use 'created' as permalink
+  $variables['created_permalinked'] = l($variables['created'], $uri['path'], $uri['options']);
   $variables['permalink'] = l(t('Permalink'), $uri['path'], $uri['options']);
   $variables['submitted'] = t('!datetime by !username', array('!datetime' => $variables['created'], '!username' => $variables['author']));
 
+  // Check to see if title should be hidden
+  if ($node_type->settings['comment_subject_hide']) {
+    $variables['title'] = "";
+  }
```

Comment:
I don't think we should do this. See issue discussion for reasoning.

Reply:
Agreed